src/middlewares.js

```diff
@@ -11,6 +11,8 @@ import rateLimit from 'express-rate-limit';
 import { RateLimitOptions } from './Options/Definitions';
 import pathToRegexp from 'path-to-regexp';
 import ipRangeCheck from 'ip-range-check';
+import RedisStore from 'rate-limit-redis';
```

Comment:
What does this means in terms of resource overhead for deployments that do not use the rate limiter or redis?

Reply:
Running `npm i --prod` and then `du -s node_modules`

`node_modules` before: 266.96mb
`node_modules` after: 269.632mb

Difference: 2.672mb

Reply:
OK, this is file size. I'm more concerned about memory.

Reply:
I’m not sure what you mean - if they haven’t defined rateLimiter.redisURL, the code will never be called

Reply:
The import statements will be called though. Depending on how the dependency is set up internally, just importing it could already have a resource impact.

Reply:
I will try to investigate

Reply:
Maybe just profiling the server init memory during start-up and start-up time itself could be enough if you do that a few of times. If there is no notable difference then let's just merge it.

Reply:
Alright, thanks for the pointer! I'll report here 😊

Reply:
Before:

![Screenshot 2023-02-01 at 1 14 16 pm](https://user-images.githubusercontent.com/4974769/215928586-7f0364b1-c157-4b9f-a28b-bcf996b5c08e.png)

Afer:

![Screenshot 2023-02-01 at 1 14 37 pm](https://user-images.githubusercontent.com/4974769/215928633-985459ed-c512-40b7-a81e-06917b423a22.png)

Reply:
Nice!